File: services/catalog/app/models/cliente.py
"""
Cliente Models para Catalog API
Baseados no schema.sql - tabela cliente
Autenticação via Microsoft OAuth apenas
"""
from pydantic import BaseModel, EmailStr
from typing import Optional


class ClienteCreate(BaseModel):
    """Model para criar cliente (POST)"""
    nome: str
    email: EmailStr
    data_expiracao: str  # ISO format: "2025-12-31"
    criado_por: str  # ID do admin que está a criar


class ClienteUpdate(BaseModel):
    """Model para atualizar cliente (PUT)"""
    nome: Optional[str] = None
    email: Optional[EmailStr] = None

class ClienteResponse(BaseModel):
    """Model para retornar cliente (GET)"""
    id: str
    nome: str
    email: str
    data_registo: str
    data_expiracao: str
    criado_por: str
    
    class Config:
        from_attributes = True

File: services/frontend/src/types/Cliente.ts
/**
 * Cliente Types
 * 
 * Baseado no schema.sql e endpoints do Catalog Service
 */

/**
 * Cliente (completo)
 */
export interface Cliente {
  id: string;
  nome: string;
  email: string;
  data_registo: string;     // ISO datetime
  data_expiracao: string;   // ISO datetime
  criado_por: string;       // ID do admin
  criado_em: string;        // ISO datetime
  atualizado_em: string;    // ISO datetime
}

/**
 * Cliente Create (POST /api/clientes/)
 */
export interface ClienteCreate {
  nome: string;
  email: string;
  password: string;         // Será hashado no backend
  data_registo: string;     // ISO datetime
  data_expiracao: string;   // ISO datetime
  criado_por: string;       // ID do admin que está criando
}

/**
 * Cliente Update (PUT /api/clientes/{id})
 * Todos os campos são opcionais (partial update)
 */
export interface ClienteUpdate {
  nome?: string;
  email?: string;
  password?: string;
  data_registo?: string;
  data_expiracao?: string;
}

/**
 * Cliente Response (alias para Cliente)
 */
export type ClienteResponse = Cliente;

/**
 * Cliente com status calculado (para frontend)
 */
export interface ClienteComStatus extends Cliente {
  status: "ativo" | "expira_breve" | "expirado" | "futuro";
  dias_restantes?: number;
}

/**
 * Cliente Stats (para analytics)
 */
export interface ClienteStats {
  cliente_id: string;
  nome: string;
  email: string;
  total_logins: number;
  total_demos_abertas: number;
  demos_unicas: number;
  ultima_atividade: string | null;
}
